lib/api_watch_diff.ml

```diff
@@ -1,12 +1,75 @@
-let diff_interface ~reference ~current =
-  let typing_env = Env.empty in
-  let coercion1 () =
-    Includemod.signatures typing_env ~mark:Mark_both reference current
+open Types
+
+type change = Added | Removed | Modified
+type diff = Value of string * change | Any
+
+module FieldMap = Map.Make (struct
+  type t = string
+
+  let compare = String.compare
+end)
+
+let env_setup ref_sig curr_sig =
```

Comment:
When there are several arguments with the same type, here both are `signature`, it's usually good practice to use labeled arguments to avoid confusion when reading code and errors when writing it!